hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskScheduler.java

```diff
@@ -196,17 +196,7 @@ protected void remove(Id id) {
         this.call(() -> {
             // Construct vertex from task
             HugeVertex vertex = this.tx().constructVertex(task);
-            // Delete the old record if exist
-            Iterator<Vertex> old = this.tx().queryVertices(vertex.id());
-            if (old.hasNext()) {
-                HugeVertex oldV = (HugeVertex) old.next();
-                assert !old.hasNext();
-                if (this.tx().indexValueChanged(oldV, vertex)) {
-                    // Only delete vertex if index value changed else override
-                    this.tx().removeVertex(oldV);
-                }
-            }
-            // Do update
+            // Add or update task info in backend store
```

Comment:
create a method updateIndex(HugeVertex vertex) for deleting code
and move to TaskTx